rust/extractor/src/translate.rs

```diff
@@ -64,6 +69,77 @@ impl CrateTranslator<'_> {
         })
     }
 
+    fn emit_location_ast_ptr<T>(
```

Comment:
minor bike-shedding naming nit: maybe here and in the following `emit_ast_ptr_location`, or `emit_location_for_ast_ptr`? Like it is it could be read "emit the ast_ptr for a location"